src/framework/main_control_loop.py

```diff
@@ -1,20 +1,30 @@
-from ..ControlTasks import ControlTaskBase, ClockManager, MissionManager, ReadCamera, PointTracker, ProcessFrame, DisplayFrame
+from ..ControlTasks import (ControlTaskBase, ClockManager, MissionManager,
+                            Webcam, PointTracker, ProcessFrame, DisplayFrame,
+                            DepthCamera, StaticFrame)
 from ..sfr import StateFieldRegistry
 import time
 
 
 class MainControlLoop(ControlTaskBase):
     def setup(self):
         self.sfr = StateFieldRegistry()
-
         """All the setup required for the MainControlLoop."""
         self.clock_manager = ClockManager(self.config, self.sfr)
-        self.read_camera = ReadCamera(self.config, self.sfr)
+
+        # set how to read images depending on run mode
+        if self.config['run_mode'] == 'HOOTL':
+            self.read_camera = StaticFrame(self.config, self.sfr)
+        elif self.config['run_mode'] == 'HITL':
+            if self.config['camera'] == 'webcam':
+                self.read_camera = Webcam(self.config, self.sfr)
+            elif self.config['camera'] == 'depthcam':
+                self.read_camera = DepthCamera(self.config, self.sfr)
+
```

Comment:
let's add a NameAssertion on the else branches for all of these so that if someone types something wrong we fail hard.